sql/core/src/test/resources/sql-tests/inputs/pgSQL/with.sql

```diff
@@ -0,0 +1,1222 @@
+--   y(id) AS
+--     (VALUES (1) UNION ALL SELECT id+1 FROM y WHERE id < 10)
+-- SELECT y.*, x.* FROM y LEFT JOIN x USING (id);
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--   x(id) AS
+--     (VALUES (1) UNION ALL SELECT id+1 FROM x WHERE id < 5),
+--   y(id) AS
+--     (VALUES (1) UNION ALL SELECT id+1 FROM x WHERE id < 10)
+-- SELECT y.*, x.* FROM y LEFT JOIN x USING (id);
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--   x(id) AS
+--     (SELECT 1 UNION ALL SELECT id+1 FROM x WHERE id < 3 ),
+--   y(id) AS
+--     (SELECT * FROM x UNION ALL SELECT * FROM x),
+--   z(id) AS
+--     (SELECT * FROM x UNION ALL SELECT id+1 FROM z WHERE id < 10)
+-- SELECT * FROM z;
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--   x(id) AS
+--     (SELECT 1 UNION ALL SELECT id+1 FROM x WHERE id < 3 ),
+--   y(id) AS
+--     (SELECT * FROM x UNION ALL SELECT * FROM x),
+--   z(id) AS
+--     (SELECT * FROM y UNION ALL SELECT id+1 FROM z WHERE id < 10)
+-- SELECT * FROM z;
+
+--
+-- Test WITH attached to a data-modifying statement
+--
+
+-- [ORIGINAL SQL]
+--CREATE TEMPORARY TABLE y (a INTEGER) USING parquet;
+CREATE TABLE y (a INTEGER) USING parquet;
+-- [ORIGINAL SQL]
+--INSERT INTO y SELECT generate_series(1, 10);
+INSERT INTO y SELECT EXPLODE(SEQUENCE(1, 10));
+
+-- [SPARK-28147] Support RETURNING clause
+--WITH t AS (
+--	SELECT a FROM y
+--)
+--INSERT INTO y
+--SELECT a+20 FROM t RETURNING *;
+--
+--SELECT * FROM y;
+
+-- [NOTE] Spark SQL doesn't support UPDATE statement
+----WITH t AS (
+----	SELECT a FROM y
+----)
+----UPDATE y SET a = y.a-10 FROM t WHERE y.a > 20 AND t.a = y.a RETURNING y.a;
+--
+--SELECT * FROM y;
+
+-- [NOTE] Spark SQL doesn't support DELETE statement
+--WITH RECURSIVE t(a) AS (
+--	SELECT 11
+--	UNION ALL
+--	SELECT a+1 FROM t WHERE a < 50
+--)
+--DELETE FROM y USING t WHERE t.a = y.a RETURNING y.a;
+--
+--SELECT * FROM y;
+
+DROP TABLE y;
+
+--
+-- error cases
+--
+
+-- INTERSECT
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE x(n) AS (SELECT 1 INTERSECT SELECT n+1 FROM x)
+--	SELECT * FROM x;
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE x(n) AS (SELECT 1 INTERSECT ALL SELECT n+1 FROM x)
+--	SELECT * FROM x;
+
+-- EXCEPT
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE x(n) AS (SELECT 1 EXCEPT SELECT n+1 FROM x)
+--	SELECT * FROM x;
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE x(n) AS (SELECT 1 EXCEPT ALL SELECT n+1 FROM x)
+--	SELECT * FROM x;
+
+-- no non-recursive term
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE x(n) AS (SELECT n FROM x)
+--	SELECT * FROM x;
+
+-- recursive term in the left hand side (strictly speaking, should allow this)
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE x(n) AS (SELECT n FROM x UNION ALL SELECT 1)
+--	SELECT * FROM x;
+
+-- [ORIGINAL SQL]
+--CREATE TEMPORARY TABLE y (a INTEGER);
+CREATE TABLE y (a INTEGER) USING parquet;
```

Comment:
Please keep `TEMPORARY`.

Reply:
I think `INSERT INTO` follows this one too.